sympy/core/symbol.py

```diff
@@ -577,7 +581,8 @@ def matches(self, expr, repl_dict=None, old=False):
 
 _range = _re.compile('([0-9]*:[0-9]+|[a-zA-Z]?:[a-zA-Z])')
 
-def symbols(names, *, cls=Symbol, **args):
+
+def symbols(names, *, cls=Symbol, **args) -> Any:
```

Comment:
Does this do anything? And couldn't it be `Union[Symbol,Tuple[Symbol]]`?

Reply:
It's probably not possible to type this but really the type is the return value of the `cls` parameter (or a tuple of the same).

Reply:
`Any` forces mypy to check function body. Apparently mypy doesn’t complain this function on master. But if you add `Any`, it will.

Reply:
Having Union as return type of not-overloaded function always makes trouble, as at call site `isinstanceof` has to be inserted.
I personally hope the `OneOf` type proposed (here)[https://github.com/python/typing/issues/1096#issuecomment-1055673065] could be added to standard and we can have union return type here.

Reply:
> Any forces mypy to check function body. Apparently mypy doesn’t complain this function on master. But if you add Any, it will.

I would have expected the opposite, that without `Any` it tries to do type inference on the body, but with `Any` it takes that as a signal to not try type inference.